string_manipulation.rb

```diff
@@ -1,30 +1,127 @@
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  puts "NOT IMPLEMENTED"
+  length = my_string.length
+  i = 0
+
+  if length >= 2
+  (length / 2).times do
+    temp = my_string[i]
+    my_string[i] = my_string[length - 1]
+    my_string[length - 1] = temp
+    i += 1
+    length -= 1
+    end
+  end
+  return my_string
 end
 
 # A method to reverse each word in a sentence, in place.
 def reverse_words(my_words)
-  puts "NOT IMPLEMENTED"
+  length = my_words.length
+
+  word_start = 0
+  word_end = 0
+
+  length.times do |i|
+    if my_words[i] == " " || i == length - 1
+
+      if my_words[i] == " "
+        word_end = i - 1
+      else
+        word_end = i
+      end
+
+      my_words[word_start..word_end] = string_reverse(my_words[word_start..word_end])

```

Comment:
The moment you do you my_words[words_start..word_end] Ruby creates a new object (aka more storage). Try it out in irb:
`temp = "something"`
`temp.object_id`
`temp[2..5].object_id`
The object_ids will show as different. 
You can work around this by creating a helper method which takes starting index and ending index along with the string and reverses the characters from starting index to ending index (just like string_reverse only bound by the indexes you send as parameters instead of 0 and length-1.